src/analysis/processing/pdal/qgspdalalgorithmbase.cpp

```diff
@@ -288,4 +289,37 @@ QVariantMap QgsPdalAlgorithmBase::processAlgorithm( const QVariantMap &parameter
   return outputs;
 }
 
+QgsPointCloudLayer *QgsPdalAlgorithmBase::parameterAsPointCloudLayer( const QVariantMap &parameters, const QString &name, QgsProcessingContext &context, QgsProcessing::LayerOptionsFlags flags ) const
+{
+  QgsPointCloudLayer *layer = QgsProcessingParameters::parameterAsPointCloudLayer( parameterDefinition( name ), parameters, context, flags );
+
+  if ( !layer )
+    return nullptr;
+
+  if ( layer->dataProvider()->name() == QStringLiteral( "copc" ) )
+  {
+    return layer;
+  }
+
+  const QString copcFileName = QgsPdalAlgorithmBase::copcIndexFile( layer->source() );
```

Comment:
How does it work with remote sources (going through http/https) ? I guess we could just use the remote URL as it is and not try to modify it, to avoid trying some non-sense file names...  (see e.g. QgsCopcPointCloudIndex::load() how it differentiates local/remote files)

Reply:
update